Sources/HttpPipeline/SharedMiddlewareTransformers.swift

```diff
@@ -160,7 +160,7 @@ public func requireHttps<A>(allowedInsecureHosts: [String])
 
 public func validateBasicAuth(user: String, password: String, request: URLRequest) -> Bool {
 
-  let auth = request.allHTTPHeaderFields?.first(where: { $0.key == "Authorization" })?.value ?? ""
+  let auth = request.allHTTPHeaderFields?.first(where: { $0.key.lowercased() == "authorization" })?.value ?? ""
```

Comment:
One change we could make is to use the `URLRequest` API for fetching this value, which appears to not worry about case.

``` swift
req.addValue("hello", forHTTPHeaderField: "Authorization")
req.value(forHTTPHeaderField: "authorization") // "hello"
```

```suggestion
  let auth = request.value(forHTTPHeaderField: "Authorization") ?? ""
```

Reply:
looks much nicer!